From 7c8a410a7f85d271b83dc5eda4cbab0a6bcd5c48 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20De=20Canni=C3=A8re?= <olivier.decanniere@qt.io>
Date: Fri, 26 Sep 2025 10:16:47 +0200
Subject: [PATCH 165/239] Extending QML: Clarify the need for qmldir files with
 qmake/cmake

The fact that a qmlfir file needs to be written manually when using
qmake was not really mentioned before.

Pick-to: 6.8 6.5
Change-Id: I174ade01d5e8bd28626939f7257717070c4d220f
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
Reviewed-by: Leena Miettinen <riitta-leena.miettinen@qt.io>
(cherry picked from commit f2c7a26b816b5e6d8cfd8e989a9e34690711cd42)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qml/doc/src/cppintegration/extending-tutorial.qdoc | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/qml/doc/src/cppintegration/extending-tutorial.qdoc b/src/qml/doc/src/cppintegration/extending-tutorial.qdoc
index 3082c70020..cf83e92760 100644
--- a/src/qml/doc/src/cppintegration/extending-tutorial.qdoc
+++ b/src/qml/doc/src/cppintegration/extending-tutorial.qdoc
@@ -101,13 +101,21 @@ For the registration to take effect, the \c qmltypes option is added to
 
 \snippet tutorials/extending-qml/chapter1-basics/chapter1-basics.pro 0
 
+Additionally, a \l{Module Definition qmldir Files}{qmldir} file needs to be added manually to
+create a \l{Writing QML Modules}{QML module}.
+
+\quotefile tutorials/extending-qml/chapter1-basics/qmldir
+
 \section2 CMake Setup
 
-Similarly, for the registration to take effect when using CMake, use the
+For the registration to take effect when using CMake, use the
 \l{qt6_add_qml_module} {qt_add_qml_module} command:
 
 \snippet tutorials/extending-qml/chapter1-basics/CMakeLists.txt 0
 
+The \l{qt6_add_qml_module}{qt_add_qml_module} API automatically generates a
+\c qmldir file for the QML module.
+
 \section2 Class Implementation
 
 The class implementation in \c piechart.cpp simply sets and returns the
-- 
2.51.2

